### PR TITLE
Avoid errors when generating the logout URL when there is no firewall key

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #15853 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Bundle/SecurityBundle/DataCollector/SecurityDataCollector.php [hunk 1]
```diff
@@ -34,10 +36,11 @@ class SecurityDataCollector extends DataCollector
      * @param TokenStorageInterface|null  $tokenStorage
      * @param RoleHierarchyInterface|null $roleHierarchy
      */
-    public function __construct(TokenStorageInterface $tokenStorage = null, RoleHierarchyInterface $roleHierarchy = null)
+    public function __construct(TokenStorageInterface $tokenStorage = null, RoleHierarchyInterface $roleHierarchy = null, LogoutUrlGenerator $logoutUrlGenerator = null)
```
Comment (Tobion) [inline src/Symfony/Bundle/SecurityBundle/DataCollector/SecurityDataCollector.php]: phpdoc missing

File: src/Symfony/Bundle/SecurityBundle/DataCollector/SecurityDataCollector.php [hunk 2]
```diff
@@ -78,11 +82,19 @@ public function collect(Request $request, Response $response, \Exception $except
                     }
                 }
             }
+
+            try {
+                $logoutUrl = $this->logoutUrlGenerator->getLogoutPath();
```
Comment (Tobion) [inline src/Symfony/Bundle/SecurityBundle/DataCollector/SecurityDataCollector.php]: this fails if logoutUrlGenerator is null

Comment (javiereguiluz) [inline src/Symfony/Bundle/SecurityBundle/DataCollector/SecurityDataCollector.php]: Right. Fixed. Thanks.

**Comment by weaverryan** [comment]
Adding the optional constructor arg is consistent with our BC policy, but should be minimized. But since this is a data collector, it seems safe to me. As does swallowing the `\Exception`, which would normally be a problem.

Unless someone has an issue with the added constructor arg, :+1: 

Status: Reviewed

**Comment by fabpot** [comment]
Thank you @javiereguiluz.

**Pull request closed without merging**